### PR TITLE
[ShadowBluetoothDevice] Override the constructor for BluetoothDevice …

### PR DESCRIPTION
**Pull request by brettchabot**
…to allow object creation even when getService() throws an Exception.

PiperOrigin-RevId: 156191900

### CODE REVIEW

File: robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowBluetoothDevice.java [hunk 1]
```diff
@@ -3,12 +3,27 @@
...
+import org.robolectric.util.ReflectionHelpers;
 
 @Implements(BluetoothDevice.class)
 public class ShadowBluetoothDevice {
 
+  @RealObject private BluetoothDevice realObject;
   private String name;
 
+  public void __constructor__(String address) {
+    // Call getService() like the original constructor would, but ignore any Exceptions. This allows
```
Comment (xian) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowBluetoothDevice.java]: If this were a javadoc comment it'd automatically appear in the Android developer docs with the Robolectric plugin.

File: robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowBluetoothDevice.java [hunk 2]
```diff
@@ -3,12 +3,27 @@
...
+import org.robolectric.annotation.RealObject;
+import org.robolectric.util.ReflectionHelpers;
 
 @Implements(BluetoothDevice.class)
 public class ShadowBluetoothDevice {
 
+  @RealObject private BluetoothDevice realObject;
   private String name;
 
+  public void __constructor__(String address) {
```
Comment (xian) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowBluetoothDevice.java]: ... and maybe it'd make more sense to shadow getService()?

Comment (jongerrish) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowBluetoothDevice.java]: Agreed!

**Comment by xian** [comment]
Could you rebase on master?

**Comment by utzcoz** [comment]
Hi @brettchabot , looks like this PR has been merged by commit https://github.com/robolectric/robolectric/commit/b6b8385e574dd61e8d45e50e80ace8d3eeb2506a. I will close this PR and delete related branch firstly.

**Pull request closed without merging**